core/rio/trig/src/main/java/org/eclipse/rdf4j/rio/trig/TriGWriter.java

```diff
@@ -100,6 +100,32 @@ public void handleStatement(Statement st)
 			throw new RuntimeException("Document writing has not yet been started");
 		}
 
+        // If we are pretty-printing, all writing is buffered until endRDF is
+        // called
```

Comment:
To avoid OoM errors, perhaps we can use a configurable buffer size instead of just waiting for endRDF; similar to what is done in  [BufferedGroupingRDFHandler](https://github.com/eclipse/rdf4j/blob/4d544bba5abba239269493ba6869a2acc6bff380/core/rio/api/src/main/java/org/eclipse/rdf4j/rio/helpers/BufferedGroupingRDFHandler.java).